routemaster/client.rb

```diff
@@ -26,48 +26,23 @@ def configure
         end
       end
 
-      def created(topic, callback, timestamp = nil, t: nil, async: false, data: nil)
-        _warn_timestamp_deprecation(timestamp)
-        _send_event('create', topic, callback, t: t || timestamp, async: async, data: data)
-      end
-
-      def created_async(topic, callback, timestamp = nil, t: nil, data: nil)
-        _warn_timestamp_deprecation(timestamp)
-        _warn_async_deprecation
-        _send_event('create', topic, callback, t: t || timestamp, async: true, data: data)
-      end
-
-      def updated(topic, callback, timestamp = nil, t: nil, async: false, data: nil)
-        _warn_timestamp_deprecation(timestamp)
-        _send_event('update', topic, callback, t: t || timestamp, async: async, data: data)
-      end
-
-      def updated_async(topic, callback, timestamp = nil, t: nil, data: nil)
-        _warn_timestamp_deprecation(timestamp)
-        _warn_async_deprecation
-        _send_event('update', topic, callback, t: t || timestamp, async: true, data: data)
-      end
-
-      def deleted(topic, callback, timestamp = nil, t: nil, async: false, data: nil)
-        _warn_timestamp_deprecation(timestamp)
-        _send_event('delete', topic, callback, t: t || timestamp, async: async, data: data)
-      end
-
-      def deleted_async(topic, callback, timestamp = nil, t: nil, data: nil)
-        _warn_timestamp_deprecation(timestamp)
-        _warn_async_deprecation
-        _send_event('delete', topic, callback, t: t || timestamp, async: true, data: data)
-      end
-
-      def noop(topic, callback, timestamp = nil, t: nil, async: false, data: nil)
-        _warn_timestamp_deprecation(timestamp)
-        _send_event('noop', topic, callback, t: t || timestamp, async: async, data: data)
-      end
+      # define event factory methods
+      {
+        created: 'create',
+        updated: 'update',
+        deleted: 'delete',
+        noop: 'noop'
+      }.each do |method_name, event_type|
+        define_method(method_name) do |topic, callback, timestamp = nil, t: nil, async: false, **rest|
```

Comment:
Why `**rest`? You're changing the method prototype. Any other named argument but `data:` and (now) `target:` would be invalid, but raise an exeption in the "wrong" method (`_send_event` instead of the currently defined one).

Reply:
(not a huge deal though)

Reply:
I think you are right. It was just getting too long and only passed through.